strawberryfields/cli/__init__.py

```diff
@@ -0,0 +1,219 @@
+
+    Related arguments:
+    * token: the authentication token to use
+    * local: whether or not to create the configuration file locally
+
+    Args:
+        args (ArgumentParser): arguments that were specified on the command
+            line stored as attributes in an argument parser object
+    """
+    if args.token:
+        kwargs = {"authentication_token": args.token}
+    else:
+        kwargs = configure_everything()
+
+    if args.local:
+        store_account(**kwargs, location="local")
+    else:
+        store_account(**kwargs)
+
+
+def ping():
+    """Tests the connection to the remote backend."""
+    if Connection().ping():
+        sys.stdout.write("You have successfully authenticated to the platform!\n")
+    else:
+        sys.stdout.write("There was a problem when authenticating to the platform!\n")
+    sys.exit()
+
+
+def configure_everything():
+    """Provides an interactive selection wizard on the command line to
+    configure every option for the API connection.
+
+    Default configuration options are provided as defaults to the user.
+    Configuration options as detailed in :doc:`/introduction/configuration`.
+
+    Returns:
+        dict[str, Union[str, bool, int]]: the configuration options
+    """
+    default_config = create_config()["api"]
+
+    authentication_token = (
+        input(PROMPTS["authentication_token"].format(default_config["authentication_token"]))
+    )
+
+    if not authentication_token:
+        sys.stdout.write("No authentication token was provided, please configure again.")
+        sys.exit()
+
+    hostname = (
+        input(PROMPTS["hostname"].format(default_config["hostname"])) or default_config["hostname"]
+    )
+
+    ssl_default = "y" if default_config["use_ssl"] else "n"
+    ssl_input = input(PROMPTS["use_ssl"].format(ssl_default)) or ssl_default
+    use_ssl = ssl_input.upper() == "Y"
+
+    port = input(PROMPTS["port"].format(default_config["port"])) or default_config["port"]
+
+    kwargs = {
+        "authentication_token": authentication_token,
+        "hostname": hostname,
+        "use_ssl": use_ssl,
+        "port": port,
+    }
+    return kwargs
+
+def run_blackbird_script(args):
+    """Run a blackbird script.
+
+    Related arguments:
+    * input: the input blackbird script to be run
+    * output: the output file to store the results in (optional)
+
+    Args:
+        args (ArgumentParser): arguments that were specified on the command
+            line stored as attributes in an argument parser object
+    """
+    try:
+        program = load(args.input)
+    except FileNotFoundError:
+        sys.stdout.write("The {} blackbird script was not found.".format(args.input))
+        sys.exit()
+
+    eng = StarshipEngine(program.target)
+
+    sys.stdout.write("Executing program on remote hardware...\n")
+    result = eng.run(program)
+
+    if result and result.samples is not None:
+        write_script_results(result.samples, output_file=args.output)
```

Comment:
How come we are only writing if `result.samples` is not None? Shouldn't we always attempt to write the output, even if it is non-numeric?

Reply:
I was mostly relying on previous code here. What would you mean by the output being non-numeric?

One thing to note is that currently only `samples` get written, though this can be modified with ease. Would you have any suggestions on that? :)

Reply:
Ah, I'm just wondering if it's possible that the user could submit a job where `result.samples` is None, in which case they would be wondering why there is no output. But, thinking about it more, I don't think this will ever be the case!

Still, might be better to catch the case where result doesn't exist/result.samples is None so that a warning or error can be returned to the user, rather than nothing happening.

Reply:
Oh, that's a great idea!

Reply:
Yes, it was like this before but I think it makes sense to alert the user if the result is malformed somehow :+1: